cnd/src/db/tables.rs

```diff
@@ -189,6 +189,46 @@ impl InsertableHalight {
     }
 }
 
+#[derive(Associations, Clone, Copy, Debug, Identifiable, Queryable, PartialEq)]
+#[belongs_to(Swap)]
+#[table_name = "hbits"]
+pub struct Hbit {
+    id: i32,
+    swap_id: i32,
+    pub amount: Text<Satoshis>, // FIXME: Do we want to store this as Satoshis?
+    pub network: Text<bitcoin::Network>,
+    pub hash_function: Text<HashFunction>,
+    pub redeem_identity: Option<Text<bitcoin::PublicKey>>,
+    pub refund_identity: Option<Text<bitcoin::PublicKey>>,
+    pub ledger: Text<Ledger>,
+}
+
+#[derive(Insertable, Clone, Copy, Debug)]
+#[table_name = "hbits"]
+pub struct InsertableHbit {
```

Comment:
Is this best-practice? 
I'd assume you can reuse the same type for inserting and loading.

Reply:
They differ in the way that `Insertable` doesn't have a primary key. Not sure if you can reuse them then?

Reply:
It won't work to make this field optional?

Reply:
Please don'e try. We wasted enough time on trying to do such kind of optimisations. 

@bonomat we are not the only ones [discussing](https://www.reddit.com/r/rust/comments/9r0n1e/diesel_queryableinsertable/) this, but as far as I could see the typical way is still to use 2 different structs for insert and query.